modules/storages/app/models/storages/storage.rb

```diff
@@ -98,6 +88,24 @@ def self.shorten_provider_type(provider_type)
       end
     end
 
+    def mark_as_unhealthy(reason: nil)
+      update(health_status: 'unhealthy', health_changed_at: Time.now.utc, health_reason: reason)
+    end
+
+    def mark_as_healthy
+      update(health_status: 'healthy', health_changed_at: Time.now.utc, health_reason: nil)
+    end
+
+    def healthy?
+      mark_as_healthy if health_status.nil?
+      health_status == 'healthy'
+    end
+
+    def unhealthy?
+      mark_as_healthy if health_status.nil?
+      health_status == 'unhealthy'
+    end
+
```

Comment:
I Suggest adding a default value to the health status in the database and simplify the logic here to side effect free methods.

Reply:
I suggest `unhealthy` as the default and setting it to `health` upon completion of the setup.